specs/carousel/carousel.md

```diff
@@ -0,0 +1,258 @@
+
+### Anatomy and Appearance
+**Carousel Structure:**
+
+Note: the Carousel class will have a method that checks the attribute names and grabs the items.
+
+```html
+<template>
+    <div
+        class="carousel"
+        aria-roledescription="carousel"
+        tabindex="-1"
+    >
+    <slot name="play-toggle"> SEE DEFAULT </slot>
+    <slot name="next-flipper"> SEE DEFAULT </slot>
+    <div
+        class="carousel-items"
+        aria-live="off"
+    >
+        <!-- Solution 1 -->
+        <!-- Map all items and use css class's to display correct item based on id. -->
+        <!-- May be a less performant way as the items will be mapped each time. -->
+        ${ x => x.items.map(item, index => {
+            return `
+            <div
+                class="carousel-item ${x => x.getActiveIndex() === index ? 'active' : ''}"
+                role="group"
+                aria-roledescription="slide"
+                aria-label="${x => x.getActiveIndex()} of ${x => x.getItemCount()}"
+            >
+                ${item}
+            </div>`
+            }).join('')
+        }
+        <!-- Solution 2 -->
+        <!-- Have carousel class method that query selects this div and inserts content for the correct item based on id/index from the id's given as an array. -->
+        <div
+            class="carousel-item"
+            role="group"
+            aria-roledescription="slide"
+            aria-label="${x => x.getActiveIndex()} of ${x => x.getItemCount()}"
+        >
+            ${x => x.getActiveItem()}
+        </div>
+    </div>
+    <slot name="previous-flipper"> SEE DEFAULT </slot>
+    <slot name="tab-list">SEE DEFAULT </slot>
+    </div>
+</template>
+```
+
+**Default Flipper**
+```html
+<button
+    @click="${x => x.next()}"
+    class="next-flipper"
+>
+    <svg class="flipper-svg">
+        <path d="example path..."></path>
+    </svg>
+</button>
+```
+
+**Default Play Toggle**
+```html
+<button
+    slot="play-toggle"
+    @click="${x => x.togglePlay()}"
+    class="previous-flipper"
+>
+    <svg class="play-svg">
+        <path d="example path..."></path>
+    </svg>
+</button>
+```
+
+**Default Tab List**
+```html
+<div role="tablist" class="tablist" aria-label="A carousel of items">
+    <div tabindex="0" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+    <div tabindex="-1" role="tab" class="tablist-tab"></div>
+</div>
+```
+
+**Implementation**
+```html
+<body>
+    <fast-carousel
+        // TODO: CHECK is this a way to apply the items?
+        item-0='<img src="...image1.src">'
+        item-1='<img src="...image2.src">'
+        item-2='<img src="...image3.src">'
+    >
+        <button
+            slot="next-flipper"
+            @click="${x => x.next()}" // TODO: How do I attach the carousel.next() function to a slotted item?
```

Comment:
You could use event delegation for this. The host element would listen for click events and then dispatch an action based on which element was clicked. For example, if the composed path comes from an element that has the `[slot="next-flipper"]` attribute then we do work to render the next slide.

Reply:
Updated to have the click handler on the host element.